### PR TITLE
Consistently use futures in amethyst_assets

### PR DESCRIPTION
**Pull request by torkleyy**
Not sure if I forgot something..

### CODE REVIEW

**Comment by torkleyy** [comment]
This is very similar to #291 Problems to resolve:

* Need a way to clone boxed futures
* Specify types for `LoadError`

File: amethyst_assets/src/asset.rs [hunk 1]
```diff
@@ -51,6 +49,8 @@ where
     fn is_shared(&self) -> bool;
 }
 
+pub type AssetFuture<A> = Box<Future<Item=A, Error=BoxedErr>>;
```
Comment (omni-viral) [inline amethyst_assets/src/asset.rs]: In order to `clone` those you may add `trait FutureClone: Future + Clone`
And then wrap in in `struct AssetFuture(Box<FutureClone<Item=A, Error=BoxedErr>>)`

File: amethyst_assets/src/loader.rs [hunk 2]
```diff
@@ -6,103 +6,50 @@ use std::path::PathBuf;
...
-use {Allocator, Asset, BoxedErr, Context, Directory, Format, AssetError, LoadError, Store};
+use {Allocator, Asset, AssetFuture, BoxedErr, Context, Directory, Format, AssetError, LoadError, Store};
 
 /// Represents a future value of an asset. This future may be
 /// added to the ECS world, where the responsible system can poll it and merge
 /// it into the persistent storage once it is `Ready`.
-pub struct AssetFuture<A, E> {
-    inner: Option<Arc<AssetFutureCell<A, E>>>,
-}
+pub struct SpawnedFuture<A, E>(Receiver<Result<A, E>>);
```
Comment (omni-viral) [inline amethyst_assets/src/loader.rs]: Unused?

Comment (torkleyy) [inline amethyst_assets/src/loader.rs]: This is pretty useful for user-code in order to implement e.g. `create_context`.

Comment (omni-viral) [inline amethyst_assets/src/loader.rs]: Then it belongs to asset.rs

**Comment by omni-viral** [comment]
I've continued it in #294 

---

Review status: 0 of 7 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/amethyst/amethyst/293#-:-Ks5aoOqOhfPE2i9QUoC:b-say3ep)*

**Comment by Xaeroxe** [comment]
Handled in #294

**Pull request closed without merging**